fastlane_core/spec/feature_spec.rb

```diff
@@ -0,0 +1,137 @@
+class FeatureHelper
+  def self.disabled_class_method
+  end
+
+  def self.enabled_class_method
+  end
+
+  def disabled_instance_method
+  end
+
+  def enabled_instance_method
+  end
+
+  def self.reset_test_method
+    FeatureHelper.instance_eval { undef test_method }
+  end
+
+  def reset_test_method
+    undef test_method
+  end
+end
+
+describe FastlaneCore do
+  describe FastlaneCore::Feature do
+    describe "Register a Feature" do
+      it "registers a feature successfully with environment variable and description" do
+        expect do
+          FastlaneCore::Feature.register(env_var: "TEST_ENV_VAR", description: "Test environment variable")
+        end.not_to raise_error
+      end
+
+      it "raises an error if no environment variable specified" do
+        expect do
+          FastlaneCore::Feature.register(description: "Test environment variable")
+        end.to raise_error "Invalid Feature"
+      end
+
+      it "raises an error if no description specified" do
+        expect do
+          FastlaneCore::Feature.register(env_var: "TEST_ENV_VAR")
+        end.to raise_error "Invalid Feature"
+      end
+    end
+
+    describe '#enabled?' do
+      after do
+        ENV.delete('TEST_ENV_VAR')
+        FastlaneCore::Feature.features.clear
+      end
+
+      it "reports unregistered features as not enabled" do
+        expect(FastlaneCore::Feature.enabled?('TEST_ENV_VAR')).to be_falsey
+      end
+
+      it "reports undefined features as not enabled, even if the environment variable is set" do
+          ENV['TEST_ENV_VAR'] = '1'
+          expect(FastlaneCore::Feature.enabled?('TEST_ENV_VAR')).to be_falsey
```

Comment:
`fastlane_core` has a test helper method which could work well for you here:

```ruby
with_env_values('TEST_ENV_VAR' => '1') do
  expect(FastlaneCore::Feature.enabled?('TEST_ENV_VAR')).to be_falsey
end
```

It does the work of cleaning up the env var when the block ends, so you could remove such from the `after` block